lib/oppen.rb

```diff
@@ -94,72 +127,91 @@ def trim_trailing_whitespaces? = @trim_trailing_whitespaces
+    #
+    # The name was amusingly chosen in reference to Wadler's work on
+    # pretty printing.
+    #
     # @return [Config]
     def self.wadler(eager_print: true, trim_trailing_whitespaces: true, upsize_stack: true)
-      new(indent_anchor: IndentAnchor::ON_BEGIN, eager_print:, trim_trailing_whitespaces:, upsize_stack:)
+      new(eager_print:, indent_anchor: IndentAnchor::CURRENT_OFFSET, trim_trailing_whitespaces:, upsize_stack:)
     end
   end
 
-  # @param value [String]
-  # @param width [Integer] token width that defaults to value.length
+  # @param value [String]  the string to print.
+  # @param width [Integer] token width.
   #
-  # @return [Oppen::Token::String] a new String token
+  # @return [Token::String] a new String token.
   def self.string(value, width: value.length)
     Token::String.new(value, width:)
   end
 
-  # @see Token::Whitespace
-  #
-  # @return [Oppen::Token::Whitespace] a new Whitespace token.
+  # @return [Token::Whitespace] a new Whitespace token.
   def self.whitespace(value)
     Token::Whitespace.new(value, width: value.bytesize)
   end
 
-  # @param str [String]
-  # @param line_continuation [String] If a new line is needed display this string before the new line
-  # @param offset [Integer]
-  # @param width [Integer] token width that defaults to str.length
+  # @param str               [String]  value shown if no new line is needed.
+  # @param line_continuation [String]  if a new line is issued, this string will show just before
+  #                                    the new line.
+  # @param offset            [Integer] the additional indentation of the break.
+  # @param width             [Integer] token width.
+  #
+  # @return [Token::Break] a new Break token.
   #
-  # @return [Oppen::Token::Break] a new Break token
+  # @see Wadler#break An example of usage of line_continuation.
   def self.break(str = ' ', line_continuation: '', offset: 0, width: str.length)
     Token::Break.new(str, width:, line_continuation:, offset:)
   end
 
-  # @param line_continuation [String] If a new line is needed display this string before the new line
-  # @param offset [Integer]
+  # @param line_continuation [String]  if a new line is needed display this string at the beginning of
+  #                                    the new line, at the specified anchor position.
+  # @param offset            [Integer] the additional indentation of the break.
   #
-  # @return [Oppen::Token::LineBreak] a new LineBreak token
+  # @return [Token::LineBreak] a new LineBreak token.
+  #
+  # @see Wadler#break An example of usage of line_continuation.
   def self.line_break(line_continuation: '', offset: 0)
     Token::LineBreak.new(line_continuation:, offset:)
   end
 
-  # @param offset [Integer]
+  # @param offset [Integer] the additional indentation of the group.
+  #
+  # @return [Token::Begin] a new consistent Begin token.
   #
-  # @return [Oppen::Token::Begin] a new consistent Begin token
+  # @see Token::BreakType
```

Comment:
Is there examples there? If so, let's mention that there are examples there that clarify the break type, just like we did with `line_break`.

Reply:
Yes the examples are inside `Token::BreakType`, no need to change anything.